classwork/сlasswork2.1/TestCases.md

```diff
@@ -0,0 +1,69 @@
+### Test Case 1: First Time Run with a New Name
+
+**Steps:**
+1. Ensure there is no pre-existing history for the name "Alice".
```

Comment:
The main logic of test checks looks good but there are some problems with the way you use to describe them. It is better to have more structured way to represent them(See the following [video](https://youtu.be/sGUbyZtAtXg?t=1066&si=uNcQAeXF6cMnNRtc) as an example on how it may look like)
Therefore some of your steps are actually expected results(the ones with Verify verb). The first item is actually precondition. For steps we should have a form Step N: Action. Expected result: Some observable change(program output, files change, UI layout change, etc).
As a result your test cases can be rewritten as:

Precondition: Check folder for file. ER: file doesn't exists or no line with Alice.
Step 1. Run the program with the argument Alice using ./hello Alice. ER: program output "Welcome, Alice!"
Step 2. Run the program with the argument Alice using command ./hello Alice. ER: program output "Hello again(x2), Alice!"
Step 3. ...

Why that structure is important? After every action we have instant feedback check. As a result if the new version of the program under test failed then you automatically get steps to reproduce the issue. When we split steps and ERs then it is not clear what step caused incorrect result and you should spend extra time to analyse that. Also we should place some ready-to-use commands if possible for the step as it speeds up the test execution which is important as the same test case can be run every day or for every build especially if it is a part of smoke test and basic functionality verification.